apps/passport-server/src/services/devconnectPretixSyncService.ts

```diff
@@ -46,126 +45,126 @@ interface EventData {
-        throw new Error("Pretix Config could not be loaded");
+        throw new Error("Data failed to validate", {
+          cause: errorCause("fetching", this.organizer.id, e)
+        });
       }
 
-      logger("[DEVCONNECT PRETIX] Sync start");
-      await this.sync(devconnectPretixConfig);
-      await this.semaphoreService.reload();
-      this._hasCompletedSyncSinceStarting = true;
-      logger("[DEVCONNECT PRETIX] Sync successful");
-    } catch (e) {
-      this.rollbarService?.reportError(e);
-      logger("[DEVCONNECT PRETIX] Sync failed", e);
-    }
-  }
-
-  public stop(): void {
-    if (this.timeout) {
-      clearTimeout(this.timeout);
-    }
-  }
-
-  /**
-   * Download Pretix state, and apply a diff to our state so that it
-   * reflects the state in Pretix.
-   */
-  private async sync(pretixConfig: DevconnectPretixConfig): Promise<void> {
-    return traced(NAME, "sync", async (span) => {
-      span?.setAttribute("organizers_count", pretixConfig.organizers.length);
-
-      const syncStart = Date.now();
+      try {
+        this.validate(fetchedData);
+      } catch (e) {
+        logger(
+          `[DEVCONNECT PRETIX]: Encountered error when validating fetched data for ${this.organizer.id}: ${e}`
+        );
+        this.rollbarService?.reportError(e);
 
-      const organizerSyncPromises = []; // one per organizer
-      // Iterate over organizers and set up a sync promise for each
-      for (const organizer of pretixConfig.organizers) {
-        organizerSyncPromises.push(this.syncOrganizer(organizer));
+        throw new Error("Data failed to validate", {
+          cause: errorCause("validating", this.organizer.id, e)
+        });
       }
 
       try {
-        const results = await Promise.all(organizerSyncPromises);
+        await this.save(fetchedData);
       } catch (e) {
         logger(
-          "[DEVCONNECT PRETIX] Failed to save tickets for one or more events",
-          e
+          `[DEVCONNECT PRETIX]: Encountered error when saving data for ${this.organizer.id}: ${e}`
         );
-        setError(e, span);
         this.rollbarService?.reportError(e);
+
+        throw new Error("Data failed to validate", {
+          cause: errorCause("saving", this.organizer.id, e)
+        });
       }
+    } finally {
+      this._isRunning = false;
+    }
+  }
 
-      const syncEnd = Date.now();
+  public cancel(): void {
+    this.pretixAPI.cancelPendingRequests();
+  }
 
-      logger(
-        `[DEVCONNECT PRETIX] Sync end. Completed in ${Math.floor(
-          (syncEnd - syncStart) / 1000
-        )} seconds`
-      );
-    });
+  /**
+   * Fetch data for each of the organizer's events.
+   */
+  private async fetchData(): Promise<FetchResult[]> {
+    const fetchedData = [];
+    for (const event of this.organizer.events) {
+      fetchedData.push({
+        event,
+        data: await this.fetchEventData(this.organizer, event)
+      });
+    }
+
+    return fetchedData;
   }
```

Comment:
surround this with a `traced` where name is `OrganizerSync`, and add relevant attributes to the `span`